### PR TITLE
migrate sentry to config-based cdk

### PR DESCRIPTION
**Pull request by girarda**
## What
Migrate the Sentry connector to config-based

## How
* Replace custom source with a source inheriting from YamlDeclarativeSource
* Define the connector config in `airbyte-integrations/connectors/source-sentry/source_sentry/sentry.yaml`
* Set additionalProperties to true in the spec

## Recommended reading order
1. `airbyte-integrations/connectors/source-sentry/source_sentry/sentry.yaml`
2. `airbyte-integrations/connectors/source-sentry/source_sentry/source.py`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-sentry/source_sentry/sentry.yaml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    authenticator:
+      type: "BearerAuthenticator"
+      api_token: "{{ config.auth_token }}"
+  paginator:
+    type: LimitPaginator
+    url_base: "*ref(definitions.requester.url_base)"
+    page_size: "*ref(definitions.page_size)"
+    limit_option:
+      inject_into: "request_parameter"
+      field_name: ""
```
Comment (girarda) [inline airbyte-integrations/connectors/source-sentry/source_sentry/sentry.yaml:25]: a bit absurd to to need this... cursor pagination should be a first-class paginator

File: airbyte-integrations/connectors/source-sentry/source_sentry/sentry.yaml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+      transform: "_"
+  requester:
+    type: HttpRequester
+    name: "{{ options['name'] }}"
+    url_base: "https://{{ config.hostname }}/api/0/"
+    http_method: "GET"
+    authenticator:
+      type: "BearerAuthenticator"
+      api_token: "{{ config.auth_token }}"
+  paginator:
```
Comment (girarda) [inline airbyte-integrations/connectors/source-sentry/source_sentry/sentry.yaml:19]: the test account doesn't have enough data to test the pagination...

Comment (sherifnada) [reply airbyte-integrations/connectors/source-sentry/source_sentry/sentry.yaml:19]: what about our actual sentry account?

File: airbyte-integrations/connectors/source-sentry/source_sentry/sentry.yaml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    type: LimitPaginator
+    url_base: "*ref(definitions.requester.url_base)"
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-sentry/source_sentry/sentry.yaml:21]: why are we passing url_base to the paginator?

File: airbyte-integrations/connectors/source-sentry/source_sentry/sentry.yaml [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      stop_condition: "{{ headers.link.next.results != 'true' }}"
+  retriever:
+    type: SimpleRetriever
+    name: "{{ options['name'] }}"
+    primary_key: "{{ options['primary_key'] }}"
+
+streams:
+  - type: DeclarativeStream
+    $options:
+      name: "events"
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-sentry/source_sentry/sentry.yaml:40]: could you retain the docstrings from python like in the suggestion? 

```suggestion
      # https://docs.sentry.io/api/events/list-a-projects-events/
      name: "events"
```

File: airbyte-integrations/connectors/source-sentry/source_sentry/sentry.yaml [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    schema_loader:
+      $ref: "*ref(definitions.schema_loader)"
+    retriever:
+      $ref: "*ref(definitions.retriever)"
+      record_selector:
+        $ref: "*ref(definitions.selector)"
+      requester:
+        $ref: "*ref(definitions.requester)"
+        path: "projects/"
+      paginator:
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-sentry/source_sentry/sentry.yaml:88]: this seems incorrect? https://docs.sentry.io/api/projects/list-your-projects/ has paginatino

File: airbyte-integrations/connectors/source-sentry/source_sentry/sentry.yaml [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    authenticator:
+      type: "BearerAuthenticator"
+      api_token: "{{ config.auth_token }}"
+  paginator:
+    type: LimitPaginator
+    url_base: "*ref(definitions.requester.url_base)"
+    page_size: "*ref(definitions.page_size)"
+    limit_option:
+      inject_into: "request_parameter"
+      field_name: ""
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-sentry/source_sentry/sentry.yaml:25]: what does it mean to have an empty field_name? is this correct?

**Review by sherifnada — CHANGES_REQUESTED** [review]
pretty easy to read these YAML connectors so far. 

left a request for change because I think pagination is not implemented.

File: airbyte-integrations/connectors/source-sentry/source_sentry/sentry.yaml (hunk 3 above)
Comment (girarda) [reply airbyte-integrations/connectors/source-sentry/source_sentry/sentry.yaml:21]: a bit of an implementation detail, but it's to replace the base url from the cursor when updating the path. although maybe that could be done in the retriever. left a [comment on this](https://github.com/airbytehq/airbyte/issues/15575#issuecomment-1212541353)

File: airbyte-integrations/connectors/source-sentry/source_sentry/sentry.yaml (hunk 6 above)
Comment (girarda) [reply airbyte-integrations/connectors/source-sentry/source_sentry/sentry.yaml:25]: the right solution here is probably for `limit_option` and `page_token_option` to be optional. [left a comment](https://github.com/airbytehq/airbyte/issues/15575#issuecomment-1212543875)

File: airbyte-integrations/connectors/source-sentry/source_sentry/sentry.yaml (hunk 5 above)
Comment (girarda) [reply airbyte-integrations/connectors/source-sentry/source_sentry/sentry.yaml:89]: good catch. also added it to the issues stream

File: airbyte-integrations/connectors/source-sentry/source_sentry/sentry.yaml (hunk 2 above)
Comment (girarda) [reply airbyte-integrations/connectors/source-sentry/source_sentry/sentry.yaml:19]: Good idea! I was able to test the pagination on the events stream. The same pagination is used for issues and projects. 

`project_details` does not have pagination https://docs.sentry.io/api/projects/retrieve-a-project/

**Review by sherifnada — APPROVED** [review]

**Pull request merged**